presto-hive/src/main/java/com/facebook/presto/hive/security/ranger/RangerBasedAccessControl.java

```diff
@@ -0,0 +1,487 @@
+    private static Users users;
+    private final Map<String, Set<String>> userRolesMapping;
+    private final Map<String, Set<String>> userGroupsMapping;
+    private final Supplier<ServicePolicies> servicePolicies;
+    private HttpClient httpClient;
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final JsonCodec<Users> USER_INFO_CODEC = jsonCodec(Users.class);
+    private static final JsonCodec<List<String>> ROLES_INFO_CODEC = listJsonCodec(String.class);
+
+    @Inject
+    public RangerBasedAccessControl(RangerBasedAccessControlConfig config, @ForRangerInfo HttpClient httpClient)
+    {
+        requireNonNull(config, "config is null");
+        requireNonNull(config.getRangerHttpEndPoint(), "Ranger service http end point is null");
+        requireNonNull(config.getRangerHiveServiceName(), "Ranger hive service name is null");
+        this.httpClient = requireNonNull(httpClient, "httpClient is null");
+
+        try {
+            users = getUsers(config);
+            servicePolicies = memoizeWithExpiration(
+                    () -> {
+                        log.info("Refreshing Ranger Hive Policies!!");
+                        return getHiveServicePolicies(config);
+                    },
+                    config.getRefreshPeriod().toMillis(),
+                    MILLISECONDS);
+            userGroupsMapping = getUserGroupsMappings();
+            userRolesMapping = getRolesForUserList(config);
+            rangerAuthorizer = new RangerAuthorizer(servicePolicies, config);
+        }
+        catch (Exception e) {
+            log.error(e, "Exception while querying ranger service ");
+            throw new AccessDeniedException("Exception while querying ranger service ");
+        }
+    }
+
+    @VisibleForTesting
+    public RangerBasedAccessControl(RangerAuthorizer rangerAuthorizer, Map<String, Set<String>> userGroupsMapping, Map<String, Set<String>> userRolesMapping, Supplier<ServicePolicies> servicePolicies)
+    {
+        this.rangerAuthorizer = rangerAuthorizer;
+        this.userGroupsMapping = userGroupsMapping;
+        this.userRolesMapping = userRolesMapping;
+        this.servicePolicies = servicePolicies;
+    }
+
+    private ServicePolicies getHiveServicePolicies(RangerBasedAccessControlConfig config)
+    {
+        URI uri = uriBuilderFrom(URI.create(config.getRangerHttpEndPoint()))
+                .appendPath(RANGER_REST_POLICY_MGR_DOWNLOAD_URL + "/" + config.getRangerHiveServiceName())
+                .build();
+        Request request = setContentTypeHeaders(prepareGet())
+                .setUri(uri)
+                .build();
+
+        try {
+            return OBJECT_MAPPER.readValue(httpClient.execute(request, createStringResponseHandler()).getBody(), ServicePolicies.class);
+        }
+        catch (JsonProcessingException e) {
+            log.error(e, "Error fetching policies from Ranger server ");
+        }
+        return null;
```

Comment:
I checked existing security connection access control, I think we can return the exception from here. About not letting the service start, it would make other connector catalog unusable even though they should be usable one.